Development/nmos/authorization_operation.cpp

```diff
@@ -0,0 +1,2004 @@
+
+                }).then([&](pplx::task<void> finally)
+                {
+                    auto lock = model.write_lock(); // in order to update local state
+
+                    try
+                    {
+                        finally.get();
+
+                        metadata_received = true;
+                    }
+                    catch (const web::http::http_exception& e)
+                    {
+                        slog::log<slog::severities::error>(gate, SLOG_FLF) << "Authorization API metadata request HTTP error: " << e.what() << " [" << e.error_code() << "]";
+
+                        authorization_service_error = true;
+                    }
+                    catch (const web::json::json_exception& e)
+                    {
+                        slog::log<slog::severities::error>(gate, SLOG_FLF) << "Authorization API metadata request JSON error: " << e.what();
+
+                        authorization_service_error = true;
+                    }
+                    catch (const std::exception& e)
+                    {
+                        slog::log<slog::severities::error>(gate, SLOG_FLF) << "Authorization API metadata request error: " << e.what();
+
+                        authorization_service_error = true;
+                    }
+                    catch (const authorization_exception&)
+                    {
+                        slog::log<slog::severities::error>(gate, SLOG_FLF) << "Authorization API metadata request error";
+
+                        authorization_service_error = true;
+                    }
+                    catch (...)
+                    {
+                        slog::log<slog::severities::severe>(gate, SLOG_FLF) << "Authorization API metadata unexpected unknown exception";
+
+                        authorization_service_error = true;
+                    }
+                });
+                request.then([&]
+                {
+                    condition.notify_all();
+                });
+
+                // wait for the request because interactions with the Authorization API endpoint must be sequential
+                condition.wait(lock, [&] { return shutdown || authorization_service_error || metadata_received; });
+
+                cancellation_source.cancel();
+                // wait without the lock since it is also used by the background tasks
+                nmos::details::reverse_lock_guard<nmos::write_lock> unlock{ lock };
+                request.wait();
+
+                return metadata_received;
+            }
+
+            // make an asynchronously GET request over the Token Issuer(authorization server) to fetch public keys
+            bool request_token_issuer_public_keys(nmos::base_model& model, nmos::experimental::authorization_state& authorization_state, load_ca_certificates_handler load_ca_certificates, slog::base_gate& gate)
```

Comment:
How is this function different to `do_public_keys_requests`? Could the common code be factored out?